### PR TITLE
fix: Check keyboard status during snapshot instead of using WindowInsets listener

### PR DESCRIPTION
**Pull request by marandaneto**
## :bulb: Motivation and Context
https://posthog.com/questions/status-bar-color-not-updating

I recall fixing something [similar](https://github.com/getsentry/sentry-java/pull/1769) in the past as well.
Sometimes the new APIs (setOnApplyWindowInsetsListener) are not well documented on when and how to use them, this is just another weird corner case/side effect.

## :green_heart: How did you test it?
Testing the sample app with a different theme and status colors

## :pencil: Checklist

- [X] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] I updated the docs if needed.
- [ ] No breaking change or entry added to the changelog.

### CODE REVIEW

**Review by pauldambra — APPROVED** [review]
👍

File: posthog-samples/posthog-android-sample/build.gradle.kts [hunk 1]
```diff
@@ -34,6 +34,9 @@ android {
         sourceCompatibility = PosthogBuildConfig.Build.JAVA_VERSION
         targetCompatibility = PosthogBuildConfig.Build.JAVA_VERSION
     }
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
```
Comment (marandaneto) [inline posthog-samples/posthog-android-sample/build.gradle.kts:39]:
```suggestion
```
